src/main/java/net/mcreator/blockly/java/blocks/EventParameterSetBlock.java

```diff
@@ -0,0 +1,132 @@
+						.getExternalTriggers();
+				for (ExternalTrigger externalTrigger : externalTriggers) {
+					if (externalTrigger.getID().equals((procedure.getExternalTrigger()))) {
+						trigger = externalTrigger;
+						break;
+					}
+				}
+
+				if (trigger == null) {
+					master.getCompileNotes().add(new BlocklyCompileNote(BlocklyCompileNote.Type.ERROR,
+							L10N.t("blockly.errors.event_parameter_set.not_setter")));
+					return;
+				}
+
+				if (procedure.getParent().getGenerator() != null) {
+					//generator wrapper
+					GeneratorWrapper generatorWrapper = new GeneratorWrapper(procedure.getParent().getGenerator());
+
+					List<Element> elements = XMLUtil.getDirectChildren(block);
+					String value = null, parameter = null, valueType = null;
+					//values
+					for (Element element : elements) {
+						if (element.getNodeName().equals("field")) {
+							String name = element.getAttribute("name");
+							if ("eventparameter".equals(name)) {
+								parameter = element.getTextContent();
+							} else if ("value".equals(name)) {
+								//compatibility with datalist selector, dropdown and so on for later extension
+								value = element.getTextContent();
+							}
+						} else if (element.getNodeName().equals("value")) {
+							if (element.getAttribute("name").equals("value")) {
+								value = BlocklyToCode.directProcessOutputBlock(master, element);
+								valueType = BlocklyBlockUtil.getInputBlockType(element);
+							}
+						}
+					}
+					if (parameter != null && value != null && !parameter.isEmpty() && !value.isEmpty()) {
+						String needEventClass = generatorWrapper.map(parameter, "eventparameters", 0);
+						String needMethod = generatorWrapper.map(parameter, "eventparameters", 1);
+						String needTrigger = generatorWrapper.map(parameter, "eventparameters", 2);
+						if ("null".equals(needEventClass) || "null".equals(needTrigger) || "null".equals(needMethod)) {
+							return;
+						}
+						if (!trigger.getID().equals(needTrigger)) {
+							//if trigger is wrong
+							master.getCompileNotes().add(new BlocklyCompileNote(BlocklyCompileNote.Type.ERROR,
+									L10N.t("blockly.errors.event_parameter_set.invalid_trigger",
+											L10N.t("trigger." + needTrigger), L10N.t("trigger." + trigger.getID()))));
+							return;
+						}
+						if (master.getTemplateGenerator() != null) {
+							HashMap<String, Object> datamodel = new HashMap<>();
+							datamodel.put("field$parameter", parameter);
+							if (valueType != null) {
+								datamodel.put("input$value", value);
+								datamodel.put("input_id$value", valueType);
+							} else {
+								datamodel.put("field$value", value);
+							}
+							//if template need them, they will need generator.map. We may help them here.
+							datamodel.put("eventClass", needEventClass);
+							datamodel.put("method", needMethod);
+							datamodel.put("triggerName", needTrigger);
+							master.append("_"+master.getTemplateGenerator()

```

Comment:
Please run the formatter on your code. The code does not follow our formattes rules

Reply:
ok